astropy/cosmology/parameter/_converter.py

```diff
@@ -3,23 +3,43 @@
 from __future__ import annotations
 
 from collections.abc import Callable
-from typing import Any
+from typing import TYPE_CHECKING, Any, TypeAlias, TypeVar, overload
 
 import astropy.units as u
 
-__all__ = []
+if TYPE_CHECKING:
+    from astropy.cosmology import Cosmology, Parameter
 
-FValidateCallable = Callable[[object, object, Any], Any]
-_REGISTRY_FVALIDATORS: dict[str, FValidateCallable] = {}
+__all__: list[str] = []
 
+# Callable[[Cosmology, Parameter, Any], Any]
+_FValidateCallable: TypeAlias = Callable[["Cosmology", "Parameter", Any], Any]
```

Comment:
In this case many types are not valid. `_FValidateCallable` encompasses things like the float-conversion function, which won't work on a non-floatable value. It would be great to be able to specify the type, but I think `Any` is correct here.
I'm planning to refactor this anyway to follow the `converter` pattern of the `attrs` and `equinox` library and the PEP proposal. It should make all this more type stable.